src/client/communicator.go

```diff
@@ -64,41 +64,51 @@ func (this *Communicator) GetVmsData() (interface{}, error) {
 	if err != nil {
 		return nil, fmt.Errorf("getting vms data error: %s", err)
 	}
-	s, err := this.GetRegistryVms()
+	s, err := this.GetRegistryTemplatesData()
```

Comment:
This function is being called periodically by the exporter and again in parallel here.
Please check out my comment below with suggestion.

Reply:
singleton being used